SwordImportExportPlugin.inc.php

```diff
@@ -119,50 +120,52 @@ public function display($args, $request) {
 					'swordSettings'
 				);
 
-				import('lib.pkp.controllers.list.submissions.SelectSubmissionsListHandler');
-				$selectSubmissionsListHandler = new SelectSubmissionsListHandler(array(
-					'title' 	=> 'navigation.submissions',
-					'count' 	=> 100,
-					'inputName' 	=> 'articleId[]',
-					'getParams' => array(
-						'status'	=> STATUS_PUBLISHED,
-					),
-				));
-				$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
-				$selectSubmissionsConfig = $selectSubmissionsListHandler->getConfig();
-				$depositedIds = [];
-				foreach ($selectSubmissionsConfig['items'] as $item) {
-					$publishedArticle = $publishedArticleDao->getByArticleId($item['id']);
-					if ($ssi = $publishedArticle->getData("swordStatementIri")) {
-						$depositedIds[$item['id']] = $ssi;
-					}
-				}
-				$selectSubmissionsConfig = json_encode($selectSubmissionsConfig);
-
-
+				$selectSubmissionsListPanel = new \PKP\components\listPanels\PKPSelectSubmissionsListPanel(
+					'selectSubmissionsListPanel',
+					'select submissions panel',
+					[
+						'apiUrl' => $request->getDispatcher()->url(
+							$request,
+							ROUTE_API,
+							$context->getPath(),
+							'_submissions'
+						),
+						'canSelect' => true,
+						'canSelectAll' => true,
+						'count' => 100,
```

Comment:
Is it necessary to specify 100 here? (@NateWr and I chatted about this a while back; not sure if he has a recommendation from that...)

Reply:
I think it is. The list panel should offer pagination support and if there's any chance of there being 1000+ submissions we could end up with a pretty hefty performance issue -- not to mention the difficulty of scrolling past a long list to get to the submit button.

Reply:
Agreed, but I'd rather avoid the proliferation of a 100-entry list default if we can configure a default further up the class hierarchy.

Reply:
The root class has a default of `30`, which applies to all `ListPanel`s. It is extended to `100` in this case so that exporting lots of submissions can be done in larger batches.

Reply:
Ah, gotcha. :+1: